connection_configuration_test.go

```diff
@@ -0,0 +1,260 @@
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "aws-oauth-file")
+
+	_, err = LoadConnectionConfig()
+	assertNotNilF(t, err, "The error should occur")
+
+	_, ok := err.(*(fs.PathError))
+	assertTrueF(t, ok, "This error should be a path error")
+}
+
+func TestParseInt(t *testing.T) {
+	var i interface{}
+
+	i = 20
+	num, err := parseInt(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, num, 20)
+
+	i = "40"
+	num, err = parseInt(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, num, 40)
+
+	i = "wrong_num"
+	_, err = parseInt(i)
+	assertNotNilF(t, err, "should have failed")
+}
+
+func TestParseBool(t *testing.T) {
+	var i interface{}
+
+	i = true
+	b, err := parseBool(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, b, true)
+
+	i = "false"
+	b, err = parseBool(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, b, false)
+
+	i = "wrong_bool"
+	_, err = parseInt(i)
+	assertNotNilF(t, err, "should have failed")
+}
+
+func TestParseDuration(t *testing.T) {
+	var i interface{}
+
+	i = 300
+	dur, err := parseDuration(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, dur, time.Duration(5*int64(time.Minute)))
+
+	i = "30"
+	dur, err = parseDuration(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, dur, time.Duration(int64(time.Minute)/2))
+
+	i = false
+	_, err = parseDuration(i)
+	assertNotNilF(t, err, "should have failed")
+}
+
+type paramList struct {
+	testParams []string
+	values     []interface{}
+}
+
+func TestParseToml(t *testing.T) {
+	testCases := []paramList{
+		{
+			testParams: []string{"user", "password", "host", "account", "warehouse", "database",
+				"schema", "role", "region", "protocol", "passcode", "application", "token",
+				"tracing", "tmpDirPath", "clientConfigFile"},
+			values: []interface{}{"value"},
+		},
+		{
+			testParams: []string{"privatekey"},
+			values:     []interface{}{generatePKCS8StringSupress(testPrivKey)},
+		},
+		{
+			testParams: []string{"port", "maxRetryCount", "clientTimeout", "jwtClientTimeout", "loginTimeout",
+				"requestTimeout", "jwtTimeout", "externalBrowserTimeout"},
+			values: []interface{}{"300", 500},
+		},
+		{
+			testParams: []string{"ocspFailOpen", "insecureMode", "PasscodeInPassword", "validateDEFAULTParameters", "clientRequestMFAtoken",
+				"clientStoreTemporaryCredential", "disableQueryContextCache", "includeRetryReason", "disableConsoleLogin", "disableSamlUrlCheck"},
+			values: []interface{}{true, "true", false, "false"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		for _, param := range testCase.testParams {
+			for _, value := range testCase.values {
+				t.Run(param, func(t *testing.T) {
+					cfg := &Config{}
+					connectionMap := make(map[string]interface{})
+					connectionMap[param] = value
+					err := parseToml(cfg, connectionMap)
+					assertNilF(t, err, "The value should be parsed")
```

Comment:
Maybe we should add expected Configs and compare them?

Reply:
The purpose of this test is to confirm whether the specific data can be parsed or not for each configuration options. I already made the testing to validate the config at the first tree testing.